static/app/views/releases/thresholdsList/index.tsx

```diff
@@ -112,6 +112,39 @@ function ReleaseThresholdList({}: Props) {
       .concat([...envDiff].sort());
   }, [projects, selection.projects]);
 
+  const getEnvironmentsAvailableToProject = useMemo((): string[] => {
+    const selectedProjectIds = selection.projects.map(id => String(id));
+    const allEnvSet = new Set(projects.flatMap(project => project.environments));
+    // NOTE: mostly taken from environmentSelector.tsx
+    const unSortedEnvs = new Set(
+      projects.flatMap(project => {
+        /**
+         * Include environments from:
+         * the requested projects
+         * all member projects if 'my projects' (empty list) is selected.
+         * all projects if -1 is the only selected project.
+         */
+        if (
+          (selectedProjectIds.length === 1 &&
+            selectedProjectIds[0] === String(ALL_ACCESS_PROJECTS) &&
+            project.hasAccess) ||
+          (selectedProjectIds.length === 0 && project.isMember) ||
+          selectedProjectIds.includes(project.id)
+        ) {
+          return project.environments;
+        }
+
+        return [];
+      })
+    );
+    const envDiff = new Set([...allEnvSet].filter(x => !unSortedEnvs.has(x)));
+
+    // bubble the selected projects envs first, then concat the rest of the envs
+    return Array.from(unSortedEnvs)
+      .sort()
+      .concat([...envDiff].sort());
```

Comment:
Looks like this copying and sorting the `envDiff` but doesn't seem like we need to make a copy
```suggestion
      .concat(envDiff.sort());
```

Reply:
Ah - the `envDiff` is a set, so this is me splatting that into an array and sorting alphabetically